src/util/content-util.ts

```diff
@@ -135,23 +135,27 @@ export class ContentUtil {
       } catch (e) {
       }
       if ((key === 'utm_campaign') || (key === 'channel')) {
-        cData.push({ id: params[key], type: CorReleationDataType.SOURCE });
+        if (!Array.isArray(params[key]) && (params[key] && params[key] !== '')) {
+          cData.push({ id: params[key], type: CorReleationDataType.SOURCE });
+        } else {
+          // should generate error telemetry for duplicate campaign parameter
+        }
       } else {
         const chengeKeyUpperCase = key.split('_').map((elem) => {
           return (elem.charAt(0).toUpperCase() + elem.slice(1));
         });
-
-        utmParams[chengeKeyUpperCase.join('')] = decodeURIComponent(value);
+        utmParams[chengeKeyUpperCase.join('')] = value;
       }
     });
     if (Object.keys(utmParams)) {
       Object.keys(utmParams).map((key) => {
-        if (utmParams[key] !== undefined) {
+        if (!Array.isArray(utmParams[key]) && (utmParams[key] && utmParams[key] !== '')) {
```

Comment:
if (utmParams[key] && !Array.isArray(utmParams[key])) isnt this condition valid?

Reply:
yes....valid